lab4/provided/main.cc

```diff
@@ -4,13 +4,28 @@
 #include "direct_mapped.hh"
 #include "memory.hh"
 #include "processor.hh"
+#include "record_store.hh"
 
 int main(int argc, char *argv[])
 {
+    const char* recordFile = "simple.txt";
+    if (argc == 2) {
+        recordFile = argv[1];
+    } else if (argc > 2) {
+        std::cout << "Usage: cache_simulator [records file]" << std::endl;
+    }
+
     Processor p;
     Memory m(8);
+    RecordStore records(recordFile);
```

Comment:
Create this first, then pass it as a parameter to the processor.

Reply:
I just wanted to keep it consistent with the way the memory was assigned.